src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -230,41 +232,98 @@ private static (SyntaxTriviaList sameLine, SyntaxTriviaList endOfLine) SplitTrai
             ISyntaxFactsService syntaxFactsService,
             CancellationToken cancellationToken)
         {
-            foreach (var nodeOrToken in declarationSyntax.Parent.ChildNodesAndTokens().Reverse())
+            var firstVariableUsage = new Dictionary<ILocalSymbol, int>();
```

Comment:
the algorithm here got complex enough that i feel like it would benefit from an intuitive description explaining what's going on.